### PR TITLE
[NFC] Iterators: Make TabularToLLVM compose with IteratorsToLLVM.

### PR DESCRIPTION
**Pull request by ingomueller-net**
This commit makes TabularTypeConverter part of the public API of TabularToLLVM such that other conversions can reuse that converter. It also turns TabularToLLVM into a partial conversion such that it can be run before IteratorsToLLVM (i.e., with ops from that dialect present).

### CODE REVIEW

File: experimental/iterators/lib/Conversion/TabularToLLVM/TabularToLLVM.cpp [hunk 1]
```diff
@@ -174,7 +165,7 @@ void ConvertTabularToLLVMPass::runOnOperation() {
   typeConverter.addSourceMaterialization(addUnrealizedCast);
   typeConverter.addTargetMaterialization(addUnrealizedCast);
 
-  if (failed(applyFullConversion(module, target, std::move(patterns))))
```
Comment (nicolasvasilache) [inline experimental/iterators/lib/Conversion/TabularToLLVM/TabularToLLVM.cpp:177]: This line should be in a different commit ?

**Review by nicolasvasilache — APPROVED** [review]

File: experimental/iterators/lib/Conversion/TabularToLLVM/TabularToLLVM.cpp (hunk 1 above)
Comment (ingomueller-net) [reply experimental/iterators/lib/Conversion/TabularToLLVM/TabularToLLVM.cpp]: Done! Made it part of #579.
